examples/LoRaWANABP/LoRaWANABP.ino

```diff
@@ -0,0 +1,91 @@
+/*
+  LoRaABP.ino
+
+  Establish a connection on a LoRaWAN network in ABP mode.
+  Exchange data to/from the network.
+
+*/
+
+#include "LoRaWANNode.h"
+
+#define FRAME_DELAY 300000  // in ms. Every 5 minutes by default.
+
+HardwareSerial SerialLora(PA_1, PA_0);
+
+// Device address, network & application keys
+const char devAddr[] = "ef00cb01";
+const char nwkSKey[] = "abcdef0123456789abcdef0123456789";
+const char appSKey[] = "0123456789abcdef0123456789abcdef";
+
+// Data send
+char frameTx[] = "Hello world!";
+
+void setup()
+{
+  Serial.begin(9600);
+  Serial.println("-- LoRaWAN ABP sketch --");
+
+  // Enable the USI module and set the radio band.
+  while(!node.begin(&SerialLora, LORA_BAND_EU_868)) {
+    Serial.println("Lora module not ready");
+    delay(1000);
+  }
+
+  // Set the network keys of the module
+  while(!node.joinABP(devAddr, nwkSKey, appSKey)) {
+    Serial.println("joinABP failed!!");
+    delay(1000);
+  }
+
+  Serial.println("Lora module ready\n");
+
+  String str = "Device address: ";
+  node.getDevAddr(&str);
+  Serial.println(str);
+  str = "NwkSKey: ";
+  node.getNwkSKey(&str);
+  Serial.println(str);
+  str = "AppSKey: ";
+  node.getAppSKey(&str);
+  Serial.println(str);
+}
+
+void loop()
+{
+  uint8_t frameRx[MAX_PAYLOAD_LENGTH];
+  uint8_t len;
+  uint8_t port;
+
+  // Send unconfirmed data to a gateway (port 1 by default)
+  int status = node.sendFrame(frameTx, sizeof(frameTx), UNCONFIRMED);
+  if(status == LORA_SEND_ERROR) {
+    Serial.println("Send frame failed!!!");
+  } else if(status == LORA_SEND_DELAYED) {
+    Serial.println("Module busy or duty cycle (not sent)");
+  } else {
+    Serial.println("Frame sent");
+
+    // Check if data received from a gateway
+    if(node.receivedFrame(frameRx, &len, &port)) {
```

Comment:
isn't there a specified timeout to this call ?

Reply:
The timeout is done inside this method in the AT command reception mechanism.

Reply:
don't you think it may be useful to have a timeout that can be defined by the application ? 
Or is the timeout related to the LORA protocol  ? If the timeout needs to be kept inside receivedFrame, then it is needed to well describe the timeout value in the API.

Reply:
By default the driver didn't implement a timeout inside at_cmd_receive_async_event_downlink_data(). It was a blocking function so I add a timeout to exit after 5 seconds. This allows to be sure to receive an answer inside the RX1 or RX2 windows.